src/json-editor.component.ts

```diff
@@ -138,22 +139,15 @@ export class JsonEditorComponent extends AbstractTrackerComponent implements OnI
         this.onRecordChange.emit(json.toJS());
       });
     this.jsonSchemaService.setSchema(this.schema);
+    // construct enhanced sorted filtered keys map for objects in the record
+    this.keysStoreService.buildKeysMap(this._record, this.schema);
 
     // setup variables need for tab grouping.
     if (this.config.tabsConfig) {
-      this.schemaKeyToTabName = this.tabsUtilService.getSchemaKeyToTabName(this.config.tabsConfig, this.schema);
       this.tabNames = this.tabsUtilService.getTabNames(this.config.tabsConfig);
-      this.tabNameToKeys = this.tabsUtilService.getTabNameToKeys(this.keys, this.schemaKeyToTabName, this.tabNames);
-      this.tabNameToSubSchema = this.tabsUtilService.getTabNameToSubSchema(this.schema, this.schemaKeyToTabName);
-      this.tabsUtilService.tabSelectionSubject.subscribe(tabName => {
-        this.activeTabName = tabName;
-      });
+      this.tabsUtilService.tabSelectionSubject.subscribe(tabName => { this.appGlobalsService.activeTabName = tabName; });
```

Comment:
I agree, even removing the subject would be better since `$` suffix already means that it's an observable kind of thing.